### PR TITLE
Create `xpk` package with `utils` module

### PR DESCRIPTION
**Pull request by PBundyra**
## Fixes / Features
- Create `xpk` package
- Create `utils` module
- Move logic from `xpk.py` to `src/xpk/main.py`, and leave `xpk.py` as an entrypoint
- Install xpk with pip as part of GH CI/CD workflows to verify it

This PR serves as a preparatory PR for further refactor. It introduces `utils.py` module to set an example for further changes. 

## Testing / Documentation
Tested manually

- [ y ] Tests pass
- [ n ] Appropriate changes to documentation are included in the PR - no user-facing changes

### CODE REVIEW

File: xpk/xpk.py [hunk 1]
```diff
@@ -40,8 +40,8 @@
 import string
 import subprocess
 import sys
-import tempfile
 import time
+import utils
```
Comment (Obliviour) [inline xpk/xpk.py:44]: nit: curious if you think xpk_utils class would be better since it is more specific?

**Review by Obliviour — APPROVED** [review]
LGTM other than two checks, really happy to see a refactor occur. Thank you

File: pyproject.toml [hunk 2]
```diff
@@ -42,7 +42,7 @@ requires = ["setuptools>=61.0"]
 build-backend = "setuptools.build_meta"
 
 [project.scripts]
-xpk = "xpk:main"
+xpk = "xpk.xpk:main"
```
[the remark this answers is not in the document]
Comment (Obliviour) [reply pyproject.toml:45]: Ah I see you were able to do this in the build test of the next PR: https://github.com/google/xpk/pull/168#pullrequestreview-2228587822, nice!

File: xpk/xpk.py (hunk 1 above)
Comment (PBundyra) [reply xpk/xpk.py]: Good point, added

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PBundyra) [reply pyproject.toml:45]: I was able to do it manually as well, and it works

**Comment by PBundyra** [comment]
I will update and merge once we resolve [discussion around the structure and naming](https://github.com/google/xpk/pull/168#discussion_r1710068676)

**Comment by PBundyra** [comment]
@sharabiani @Obliviour PTAL

**Review by Obliviour — APPROVED** [review]
LGTM, glad we can still use "python xpk.py" flow + "pip install + xpk commands" and reorganize xpk into modules.

**Pull request merged**